src/learner.cc

```diff
@@ -527,15 +548,6 @@ class LearnerImpl : public Learner {
     auto n = tparam_.__DICT__();
     cfg_.insert(n.cbegin(), n.cend());
 
-    Args args = {cfg_.cbegin(), cfg_.cend()};
-    generic_parameters_.UpdateAllowUnknown(args);
-    gbm_->Configure(args);
-    obj_->Configure({cfg_.begin(), cfg_.end()});
-
-    for (auto& p_metric : metrics_) {
-      p_metric->Configure({cfg_.begin(), cfg_.end()});
```

Comment:
Where are the metrics configured now?

Reply:
Em .. I removed it because the metric has nothing to do with "model", but jvm and r still uses it for memory based serialization.

Reply:
Now it's saved as an attribute.  Hopefully we can port JVM and R to use the new serialization methods sooner.